packages/flutter_tools/test/general.shard/isolated/ios/native_assets_test.dart

```diff
@@ -222,43 +222,134 @@ void main() {
     BuildMode.debug,
     BuildMode.release,
   ]) {
-    testUsingContext('build with assets $buildMode',
-        overrides: <Type, Generator>{
-          FeatureFlags: () => TestFeatureFlags(isNativeAssetsEnabled: true),
-          ProcessManager: () => FakeProcessManager.list(
-                <FakeCommand>[
-                  const FakeCommand(
-                    command: <Pattern>[
-                      'lipo',
-                      '-create',
-                      '-output',
-                      '/build/native_assets/ios/bar.framework/bar',
-                      'arm64/libbar.dylib',
-                      'x64/libbar.dylib',
-                    ],
-                  ),
-                  const FakeCommand(
-                    command: <Pattern>[
-                      'install_name_tool',
-                      '-id',
-                      '@rpath/bar.framework/bar',
-                      '/build/native_assets/ios/bar.framework/bar'
-                    ],
-                  ),
-                  FakeCommand(
-                    command: <Pattern>[
-                      'codesign',
-                      '--force',
-                      '--sign',
-                      '-',
-                      if (buildMode == BuildMode.debug)
-                      '--timestamp=none',
-                      '/build/native_assets/ios/bar.framework',
-                    ],
-                  ),
-                ],
-              ),
-        }, () async {
+    testUsingContext('build with assets $buildMode', overrides: <Type, Generator>{
```

Comment:
These tests LGTM!

I think we should also exercise this behavior in an integration test:

* test/integration.shard/isolated/native_assets_test.dart

You could write a function near `addLinkHookDepedendency` that modifies the existing C code to depend on another dylib. And change the `hook/build.dart` to produce two dylibs.

This will also ensure we're testing it on the other OSes where no changes are needed.

Reply:
Testing linking in the integration tests on Windows is blocked by support for passing linker flags to MSVC through `CBuilder`.

Reply:
🙏 

We (@mosuem me) haven't found the time to look at Windows yet. If you want, you can embrace the yak shave and make a PR on that repo.

I'm also fine with filing an issue on that repo, and linking it here on the reason why it's skipped.